examples/chip-tool/commands/common/Command.cpp

```diff
@@ -35,30 +35,55 @@
 bool Command::InitArguments(int argc, char ** argv)
 {
     bool isValidCommand = false;
-    size_t argsCount    = mArgs.size();
 
-    size_t argsOptionalCount = 0;
-    for (size_t i = 0; i < argsCount; i++)
+    size_t argvExtraArgsCount = (size_t) argc;
+    size_t mandatoryArgsCount = 0;
+    size_t optionalArgsCount  = 0;
+    for (size_t i = 0; i < mArgs.size(); i++)
     {
         if (mArgs[i].optional)
         {
-            argsOptionalCount++;
+            optionalArgsCount++;
+        }
+        else
+        {
+            mandatoryArgsCount++;
+            argvExtraArgsCount--;
         }
     }
 
-    VerifyOrExit((size_t)(argc) >= (argsCount - argsOptionalCount) && (size_t)(argc) <= argsCount,
-                 ChipLogError(chipTool, "InitArgs: Wrong arguments number: %d instead of %zu", argc, argsCount));
+    VerifyOrExit((size_t)(argc) >= mandatoryArgsCount && (argvExtraArgsCount == 0 || (argvExtraArgsCount && optionalArgsCount)),
+                 ChipLogError(chipTool, "InitArgs: Wrong arguments number: %d instead of %zu", argc, mandatoryArgsCount));
 
-    for (size_t i = 0; i < (size_t) argc; i++)
+    // Initialize mandatory arguments
+    for (size_t i = 0; i < mandatoryArgsCount; i++)
     {
-        if (!InitArgument(i, argv[i]))
+        char * arg = argv[i];
+        if (!InitArgument(i, arg))
         {
             ExitNow();
         }
     }
 
-    for (size_t i = (size_t) argc; i < argsCount; i++)
+    // Initialize optional arguments
+    for (size_t i = mandatoryArgsCount; i < (size_t) argc; i += 2)
```

Comment:
Maybe add a comment that we step by 2 because for every optional arg we expect a name and value?